go/vt/mysqlctl/mysqld.go

```diff
@@ -1006,34 +1010,25 @@ func deleteTopDir(dir string) (removalErr error) {
 	return
 }
 
-// executeMysqlScript executes a .sql script from an io.Reader with the mysql
-// command line tool. It uses the connParams as is, not adding credentials.
-func (mysqld *Mysqld) executeMysqlScript(connParams *mysql.ConnParams, sql io.Reader) error {
-	dir, err := vtenv.VtMysqlRoot()
-	if err != nil {
-		return err
-	}
-	name, err := binaryPath(dir, "mysql")
-	if err != nil {
-		return err
-	}
-	cnf, err := mysqld.defaultsExtraFile(connParams)
+// executeMysqlScript executes a .sql script from an string.
+// It uses the connParams as is, not adding credentials.
+func (mysqld *Mysqld) executeMysqlScript(ctx context.Context, connParams *mysql.ConnParams, sql string) error {
+	connector := dbconfigs.New(connParams)
+	conn, err := connector.Connect(ctx)
 	if err != nil {
 		return err
 	}
-	defer os.Remove(cnf)
-	args := []string{
-		"--defaults-extra-file=" + cnf,
-		"--batch",
-		"--default-character-set=utf8mb4",
-	}
-	env, err := buildLdPaths()
+	defer conn.Close()
+
+	_, more, err := conn.ExecuteFetchMulti(sql, 0, false)
```

Comment:
This is a change in behavior, and one that's not mentioned in the function comment. We're passing 0 for maxrows, which means that if any query in the loaded script returns a row then we'll return an error:
```
vterrors.Errorf(vtrpc.Code_ABORTED, "Row count exceeded %d", maxrows)
```

I think we should use -1 (no limit) instead of 0 here. That keeps the behavior of this function unchanged.

Reply:
Also fixed!